npm_and_yarn/lib/dependabot/npm_and_yarn.rb

```diff
@@ -97,88 +124,121 @@ def self.extract_var(error_message)
 
         match_data
       end
+
+      sig do
+        params(
+          error_message: String,
+          dependencies: T::Array[Dependabot::Dependency],
+          yarn_lock: Dependabot::DependencyFile
+        ).returns(String)
+      end
+      def self.sanitize_resolvability_message(error_message, dependencies, yarn_lock)
+        dependency_names = dependencies.map(&:name).join(", ")
+        "Error whilst updating #{dependency_names} in #{yarn_lock.path}:\n#{error_message}"
+      end
     end
 
     YARN_CODE_REGEX = /(YN\d{4})/
     YARN_ERROR_CODES = T.let({
       "YN0001" => {
         message: "Exception error",
-        new_error: ->(_error, message) { Dependabot::DependabotError.new(message) }
+        handler: ->(message, _error, _params) { Dependabot::DependabotError.new(message) }
       },
       "YN0002" => {
         message: "Missing peer dependency",
-        new_error: ->(_error, message) { Dependabot::DependencyFileNotResolvable.new(message) }
+        handler: ->(message, _error, _params) { Dependabot::DependencyFileNotResolvable.new(message) }
       },
       "YN0016" => {
         message: "Remote not found",
-        new_error: ->(_error, message) { Dependabot::GitDependenciesNotReachable.new(message) }
+        handler: ->(message, _error, _params) { Dependabot::GitDependenciesNotReachable.new(message) }
       },
       "YN0020" => {
         message: "Missing lockfile entry",
-        new_error: ->(_error, message) { Dependabot::DependencyFileNotFound.new(message) }
+        handler: ->(message, _error, _params) { Dependabot::DependencyFileNotFound.new(message) }
       },
       "YN0046" => {
         message: "Automerge failed to parse",
-        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+        handler: ->(message, _error, _params) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
       },
       "YN0047" => {
         message: "Automerge immutable",
-        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+        handler: ->(message, _error, _params) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
       },
       "YN0062" => {
         message: "Incompatible OS",
-        new_error: ->(_error, message) { Dependabot::DependabotError.new(message) }
+        handler: ->(message, _error, _params) { Dependabot::DependabotError.new(message) }
       },
       "YN0063" => {
         message: "Incompatible CPU",
-        new_error: ->(_error, message) { Dependabot::IncompatibleCPU.new(message) }
+        handler: ->(message, _error, _params) { Dependabot::IncompatibleCPU.new(message) }
       },
       "YN0071" => {
         message: "NM can't install external soft link",
-        new_error: ->(_error, message) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
+        handler: ->(message, _error, _params) { Dependabot::MisconfiguredTooling.new("Yarn", message) }
```

Comment:
can we format this so that it reads better?

Reply:
We can convert them into lambda functions

Reply:
Converted and formatted all function into lambda function.